src/commands/color.ts

```diff
@@ -1,21 +1,26 @@
-import { Client, ColorResolvable, Message } from 'discord.js';
-
-exports.run = async (client: Client, msg: Message, args: string[]) => {
-  if (args === undefined || args.length === 0) {
-    const desc = module.exports.help.description;
-    const name = module.exports.help.name;
-    const usage = module.exports.help.usage;
-    msg.channel.send(`Name: ${name}\nDescription: ${desc}\nUsage: ${usage}`);
-    return;
-  }
-
-  const roleName = `USER-${msg.author!.id}`;
-  const myRole = msg.guild!.roles.cache.find((role: { name: string }) => role.name === roleName);
-  let chosenColor = args[0].toUpperCase();
+import { SlashCommandBuilder } from '@discordjs/builders';
+import { ColorResolvable, CommandInteraction, GuildMember } from 'discord.js';
+import { Command } from '../types/command';
+import { CommandBody } from '../types/command-body';
+
+const data = new SlashCommandBuilder()
+  .setName('color')
+  .setDescription('Changes the color of the user\'s name.')
+  .addStringOption(option =>
+    option
+      .setName('hex')
+      .setDescription('A hex based color code')
+      .setRequired(true)
+  );
+
+const execute: CommandBody = async (interaction: CommandInteraction) => {
+  const roleName = `USER-${interaction.member!.user.id}`;
+  const myRole = interaction.guild!.roles.cache.find((role: { name: string }) => role.name === roleName);
```

Comment:
Assertion!?

Reply:
Yeah, they are necessary sometimes for discordjs. I'll look into removing more of them in https://github.com/Mr1upMachine/rusty-bot/issues/74